packages/mobile/src/components/details-tile/MetadataItem.tsx

```diff
@@ -1,18 +1,41 @@
-import type { PropsWithChildren } from 'react'
+import type { ReactElement, ReactNode } from 'react'
 
-import { Flex, Text } from '@audius/harmony-native'
+import { Flex, Text, TextLink } from '@audius/harmony-native'
 
-type MetadataItemProps = PropsWithChildren<{
+export type MetadataItemProps = {
   label: string
-}>
+  value: string
+  url?: string
+  renderValue?: (value: string, link: ReactElement) => ReactNode
+}
+
+export const MetadataItem = (props: MetadataItemProps) => {
+  const { label, value, url, renderValue: renderValueProp } = props
+
+  const renderValue = () => {
+    const valueElement = url ? (
+      <TextLink url={url} variant='visible' size='s' strength='strong'>
+        {value}
+      </TextLink>
+    ) : (
+      <Text size='s' strength='strong'>
+        {value}
+      </Text>
+    )
+
+    if (renderValueProp) {
```

Comment:
nit: could put this if statement at the top to not initialize `valueElement` at all unless it's needed. don't think it rly matters tho